.github/workflows/functionality-tests.yml

```diff
@@ -11,23 +11,19 @@ jobs:
   build:
     name: Run code quality and funcionality tests
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        node_version:
-          - 16
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node_version }}
-        uses: actions/setup-node@v1
+      - name: Use Node.js 20.x
+        uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node_version }}
+          node-version-file: '.tool-versions'
       - name: Cache dependencies
         uses: actions/cache@v2
         with:
           path: |
             **/node_modules
-          key: ${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('**/package-lock.json') }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
```

Comment:
this key MUST change for every node version, so that we don't restore the cache created by a different version of node.
We can check if `actions/setup-node@v3` exposes the node version used, otherwise we can fallback by hashing the tool-versions file